app/services/base_service.rb

```diff
@@ -14,7 +14,19 @@ def initialize(result, message, original_error: nil)
     end
   end
 
-  class ThrottlingError < StandardError; end
+  class ThrottlingError < StandardError
+    attr_reader :provider, :code
+
+    def initialize(message = nil, provider: nil)
+      @provider = provider
+
+      super(message)
+    end
+
+    def provider_struct
+      OpenStruct.new(code: provider)
```

Comment:
We are trying to remove all reference to `OpenStruct` in the code base because it has some memory issues when using Ruby YJIT compiler.
You can use `Data` or `Struct` instead.